src/measure_theory/function/lp_space.lean

```diff
@@ -1316,6 +1316,14 @@ def Lp {α} (E : Type*) {m : measurable_space α} [measurable_space E] [normed_g
 localized "notation α ` →₁[`:25 μ `] ` E := measure_theory.Lp E 1 μ" in measure_theory
 localized "notation α ` →₂[`:25 μ `] ` E := measure_theory.Lp E 2 μ" in measure_theory
 
+/-- Prefer to inherit `add_comm_group → add_comm_monoid` over `add_submonoid_class → _`.
+
+This speeds up some unifications, especially with the `normed_group` instance a whole lot,
+since those problems involve huge terms due to all the parameters.
+-/
+instance [borel_space E] [second_countable_topology E] : add_comm_monoid (Lp E p μ) :=
+add_comm_group.to_add_comm_monoid _
```

Comment:
Would it make sense to just globally set the priority differently for `add_submonoid_class → _`?

Reply:
I'd much rather get rid of this different priority altogether (which would involve some extensive unifier modifications I assume). So by restricting this instance to the cases where it actually matters, we don't make the rest of the library too dependent on priority shenanigans.

Reply:
My point is, don't we want the `foo_class → foo` instances to be weaker than than the `foo_derived → foo` instances?
That way, lean will always produce `group.to_monoid (submonoid_class.to_group)` instead of `submonoid_class.to_monoid`, which is likely what we want because the former term can find the group instance by unification.

Reply:
I see. I'll try if the modified priorities make stuff work faster.

Reply:
Can you update the PR description to reflect this change, assuming you plan to keep it?

Reply:
Yes, indeed!